src/System.Diagnostics.Tracing/src/System/Diagnostics/Tracing/EventSource.cs

```diff
@@ -3643,7 +3672,7 @@ private static void AddProviderEnumKind(ManifestBuilder manifest, FieldInfo stat
             if (eventData == null || eventData.Length <= eventAttribute.EventId)
             {
                 EventMetadata[] newValues = new EventMetadata[Math.Max(eventData.Length + 16, eventAttribute.EventId + 1)];
-                Array.Copy(eventData, 0, newValues, 0, eventData.Length);
+                Array.Copy(eventData, newValues, eventData.Length);
```

Comment:
At some point I think it'd be nice to change over the implementation to use nameof, which has maintainability benefits, but it's not a huge deal.  There are perf benefits to passing in the explicit lower-bound of 0, as otherwise the implementation that doesn't take the lower bound has to ask the arrays for their lower bounds and then just delegates to this overload.